From 6cd16a66946cd0176869e018a9477b7c6987d12a Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Tue, 12 Nov 2019 18:59:45 +0100
Subject: [PATCH] PONRTSYS-6012: Do not change GSWIP clk for shortcut mode

The GSWIP is running at 666 MHZ also in short cut mode for us.

This mode was only used to change the clock of the switch, but this
feature was never validated, remove it.

We use the gsw_mode setting still to activate the shortcut mode in the
gswip.
---
 drivers/net/ethernet/lantiq/cqm/cqm_dev.c    | 11 +----------
 drivers/net/ethernet/lantiq/cqm/cqm_dev.h    |  1 -
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.c | 12 ------------
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.h |  7 -------
 4 files changed, 1 insertion(+), 30 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/cqm_dev.c b/drivers/net/ethernet/lantiq/cqm/cqm_dev.c
index e9a5ae6fdb84..c92549fd95ca 100644
--- a/drivers/net/ethernet/lantiq/cqm/cqm_dev.c
+++ b/drivers/net/ethernet/lantiq/cqm/cqm_dev.c
@@ -17,7 +17,6 @@ struct device_node *parse_dts(int j, void **pdata, struct resource **res,
 			      int *num_res)
 {
 	struct device_node *node = NULL, *cpu_deq_port = NULL;
-	struct device_node *ret_node = NULL, *gsw_node = NULL;
 	int idx = 0;
 	struct cqm_data *cqm_pdata = NULL;
 	unsigned int intr[MAX_NUM_INTR];
@@ -129,15 +128,7 @@ struct device_node *parse_dts(int j, void **pdata, struct resource **res,
 				 &cqm_pdata->gint_mode))
 		cqm_pdata->gint_mode = -1;
 
-	ret_node = node;
-	gsw_node = of_find_node_by_name(NULL, "gsw_core");
-	if (!gsw_node) {
-		pr_err("Unable to get node gsw_core\n");
-		goto err_free_pdata;
-	}
-	cqm_pdata->gsw_mode = 0;
-	of_property_read_u32(gsw_node, "gsw_mode", &cqm_pdata->gsw_mode);
-	return ret_node;
+	return node;
 
 err_free_pdata:
 	kfree(pdata);
diff --git a/drivers/net/ethernet/lantiq/cqm/cqm_dev.h b/drivers/net/ethernet/lantiq/cqm/cqm_dev.h
index 384814d391c7..c7cfac435146 100644
--- a/drivers/net/ethernet/lantiq/cqm/cqm_dev.h
+++ b/drivers/net/ethernet/lantiq/cqm/cqm_dev.h
@@ -48,7 +48,6 @@ struct cqm_data {
 	unsigned int pool_size_a1[MAX_NUM_POOLS];
 	u32  dq_port[MAX_CPU_DQ_PORT_N_TYPE];
 	u32 num_dq_port;
-	u32 gsw_mode;
 	int gint_mode;
 };
 #endif
diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
index 62d1dc840dda..d9c3ebb1dd24 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
@@ -4424,18 +4424,6 @@ static int cqm_prx300_probe(struct platform_device *pdev)
 #endif
 	/* Enable all the LS interrupts */
 	ls_intr_ctrl(0xFF0000, cqm_ctrl->ls);
-	/*set the Ingress port delay before enqueue*/
-	switch (pdata->gsw_mode) {
-	case SHORT_QOS_10G:
-		clk_set_rate(cqm_ctrl->cbm_clk, 400000000);
-		break;
-	case FULL_QOS_1G:
-	case SHORT_QOS_1G:
-		clk_set_rate(cqm_ctrl->cbm_clk, 250000000);
-		break;
-	};
-	dev_info(cqm_ctrl->dev, "CBM Clock: %ldHz\n",
-		 clk_get_rate(cqm_ctrl->cbm_clk));
 
 	/*setup the DMA channels*/
 	for (i = 0; i < CQM_ENQ_PORT_MAX; i++) {
diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.h b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.h
index bf721f13ee6e..11a56dfd3fcd 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.h
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.h
@@ -165,13 +165,6 @@ enum CQM_BUFF_SIZE {
 	CQM_SIZE3_BUF_SIZE,
 };
 
-enum CQM_QOS_LINK_MODE {
-	FULL_QOS_10G = 0,
-	SHORT_QOS_10G,
-	FULL_QOS_1G,
-	SHORT_QOS_1G,
-};
-
 /**************************************************
  *
  * Macros
